back/src/bsda/validation/schema.ts

```diff
@@ -268,6 +267,12 @@ export const rawBsdaSchema = z
           "Les opérations d'entreposage provisoire et groupement ne sont pas compatibles entre elles"
       });
     }
+
+    // Additionnal checks on the transporterCompanySiret
+    return isRegisteredSiretRefinement(
+      "TRANSPORTER",
+      val.transporterRecepisseIsExempted
+    )(val.transporterCompanySiret ?? "", ctx);
```

Comment:
- à mon sens, mieux vaut sortir le check `transporterRecepisseIsExempted` de la méthode partagée
- pas besoind evaleur par défaut pour `transporterCompanySiret` je crois, le cas ou il est falsy est géré en début de méthode
- nitpick, mais conceptuellement j'aurais évité de return. Meme si ca ne change pas grand chose

```suggestion
    // Additionnal checks on the transporterCompanySiret
    if (!val.transporterRecepisseIsExempted) {
      await isRegisteredSiretRefinement("TRANSPORTER")(val.transporterCompanySiret, ctx);
    }
```

Reply:
Bon bein en fait faut mettre le `return` sinon ça marche pas